x-pack/solutions/security/plugins/security_solution/common/constants.ts

```diff
@@ -493,6 +493,15 @@ export const DEFAULT_ALERT_TAGS_VALUE = [
   i18n.FURTHER_INVESTIGATION_REQUIRED,
 ] as const;
 
+/** Allow list of integrations to be available in the AI4DSOC integrations page */
+export const AI4DSOC_ALLOWED_INTEGRATIONS: string[] = [
```

Comment:
Why are we defining this at the Security top level `/common` directory? It seems it's only being used in the _integrations.tsx_ component. Will it need to be imported from anywhere else? Otherwise, please consider moving this to the same component file.

Reply:
@semd I am pretty sure it will be used elsewhere

We'll also have this same list available in the onboarding integrations section, but I think @PhilippeOberti was also needing it somewhere in alerts

Reply:
I don't believe it's used on the server side... It would make sense to put it under `public/common` but not sure why we would want to put it under `components/` as it's just a constant...

What about `public/common/constants.ts`, the file already exists...

Reply:
@PhilippeOberti 
>why we would want to put it under components/ as it's just a constant

My reasoning was: "It's something used only by components". But yes, you are right, it's not a component. 

I was trying to suggest a specific place for the new _searchAiLake_ common stuff, you'll probably need to create more _common_ things. 

Using a generic `constants.ts` to put random things from different places is an anti-pattern; it will eventually become a junk drawer that adds a ton of unnecessary stuff to many different chunks (like the `/common/constants.ts`).

What if we create a `/public/common/lib/search_ai_lake/*`?

Reply:
I have a request to reuse a component in this PR: https://github.com/elastic/kibana/pull/217905/files#r2039646307

vote for `/public/common/components` +1

Reply:
I don't like the idea of putting things in `public/common/components`, that folder is already full of components that are actually used in a single place... It's a nightmare to navigate... For me this folder should be things that are truly reusable and used in many places, almost like a step before being added to EUI...
We have extracted things to `common` too soon too many times in the past...

I'd much prefer having something like `public/common/lib/search_ai_lake` or even `public/search_ai_lake` then have `components`, `constants`, `hooks` folders under it...

Plus if we soon extract all the ai for soc logic in their own plugin, it will be easier to have everything as collocated as possible?

Reply:
I don't think there are any right and wrong answer here, I was just suggesting alternatives. I'll go with whatever you guys decide 😆

Reply:
I like the idea of `public/common/lib/search_ai_lake` (or even `public/common/search_ai_lake`, I prefer with `lib` though). And then create `components`, `constants`, `hooks`... 
This way we'll have everything related to _searchAiLake_ under a single directory, with clear ownership as well.

Reply:
public/common/lib/search_ai_lake 👍

Reply:
@semd @angorayc We could also have a package with truly reusable components, constants, hooks... 👀 
This is actually how I would have approached it in a proper monorepo... 😆

Reply:
Moved the constants, components and hooks to `public/common/lib/search_ai_lake` and updated @PhilippeOberti 's component to use my constant

Reply:
I see the entire page implementation has been moved to the shared directory. The idea was to put only shared code in the _common_ directory. For now, we only have the integration constants file (we can put it at `public/common/lib/search_ai_lake/integrations.ts`), and that's it. Someday, we may want to reuse some components or hooks, at that point we'll be able to move them to the shared directory.

However, I think that all the code that is not reused should not live in a shared directory, but inside the specific sub-plugin. Wasn't that the idea @PhilippeOberti ?

Reply:
Ahh apologies, I misunderstood. 

I'll move everything besides the constants back for now, and then wait to confirm if @angorayc is able to reuse anything

Reply:
Yup that was the idea in my head!

Reply:
@semd @angorayc updated now so that the only things in the new shared folder are the constants and the hook